ocs_ci/deployment/deployment.py

```diff
@@ -646,6 +649,109 @@ def patch_default_sc_to_non_default(self):
+
+        # STEP3: Follow these instructions to ensure that the PVCs listed in the previous step are deleted
+        # STEP 3.1: Determine the pod that is consuming the PVC.
+        pod_list = get_all_pods()
+        for pod in pod_list:
+            for pvc in pvcs_to_delete:
+                if get_pvc_name(pod) == pvc.get('metadata').get('name'):
+                    pod.delete()
+
+        # STEP 3.2: Delete the PVCs.
+        for pvc in pvc_list:
+            pvc.delete()
+
+        # STEP 4: Delete the StorageCluster object.
+        storage_cluster = ocp.OCP(kind=constants.STORAGECLUSTER, resource_name="ocs-storagecluster")
+        storage_cluster.delete()
+
+        # STEP 5: Delete the namespaces and wait till the deletion is complete.
+        oc = openshift_ops.OCP()
+        project_list = oc.get_projects()
+        for project in project_list:
+            if "test" in project.get('metadata').get('name'):
+                ocp_obj.delete_project(project.get('metadata').get('name'))
+        ocp_obj.delete_project('openshift-storge')
+
+        # STEP 6: List the storage nodes.
+        nodes_list = get_labeled_nodes('cluster.ocs.openshift.io/openshift-storage')
+
+        # STEP 7: For each node, perform the following:
+        for node in nodes_list:
+            ocp_obj.exec_oc_debug_cmd(node=node, cmd_list=["rm -rf /var/lib/rook"])
+
+        # STEP 8: Delete the storage classes with an openshift-storage provisioner listed in step 1.
+        delete_storageclasses(sc_list)
+
+        # STEP 9: Remove the taint from the storage nodes.NOT SUPPORTED YET
+        """ ocp_obj.exec_oc_cmd("adm taint nodes --all node.ocs.openshift.io/storage-") """
+
+        # STEP 10: Unlabel the storage nodes.
+        nodes_list.append(get_labeled_nodes('topology.rook.io/rack'))
+        for node in nodes_list:
+            ocp_obj.add_label(resource_name=node, label='cluster.ocs.openshift.io/openshift-storage-')
+            ocp_obj.add_label(resource_name=node, label='topology.rook.io/rack-')
+
+        # STEP 11: Remove CustomResourceDefinitions.
+        ocp_obj.exec_oc_cmd("delete crd backingstores.noobaa.io bucketclasses.noobaa.io"
+                            " cephblockpools.ceph.rook.io cephclusters.ceph.rook.io "
+                            "cephfilesystems.ceph.rook.io cephnfses.ceph.rook.io "
+                            "cephobjectstores.ceph.rook.io cephobjectstoreusers.ceph.rook.io "
+                            "noobaas.noobaa.io ocsinitializations.ocs.openshift.io  "
+                            "storageclusterinitializations.ocs.openshift.io "
+                            "storageclusters.ocs.openshift.io  --wait=true --timeout=5m"
+                            )
+
+        # STEP 12: To make sure that OpenShift Container Storage is uninstalled, verify that the openshift-storage
+        # namespace no longer exists REALLY NECESSARY?
+        if not ocp_obj.wait_for_delete("openshift-storage"):
+            return False
```

Comment:
should uninstall_osc() return bool? 
I think it would be better to assert here

Reply:
by my comment in the code I think this part is unnecessary at all, because in step 5 I delete the exact same thing. maybe just move the wait_for_delete() where I delete it?

Reply:
yes